geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterStore.scala

```diff
@@ -0,0 +1,366 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.raster.data
+
+import java.io.Serializable
+import java.util.Map.Entry
+import java.util.concurrent.{Callable, TimeUnit}
+import java.util.{Map => JMap}
+
+import com.google.common.cache.CacheBuilder
+import com.google.common.collect.ImmutableSetMultimap
+import com.typesafe.scalalogging.slf4j.Logging
+import org.apache.accumulo.core.client.{BatchWriterConfig, Connector, TableExistsException}
+import org.apache.accumulo.core.data.{Key, Mutation, Range, Value}
+import org.apache.accumulo.core.security.TablePermission
+import org.geotools.coverage.grid.GridEnvelope2D
+import org.joda.time.DateTime
+import org.locationtech.geomesa.accumulo.index.Strategy._
+import org.locationtech.geomesa.accumulo.iterators.BBOXCombiner._
+import org.locationtech.geomesa.accumulo.stats.{RasterQueryStat, RasterQueryStatTransform, StatWriter}
+import org.locationtech.geomesa.accumulo.util.{SelfClosingBatchScanner, SelfClosingScanner}
+import org.locationtech.geomesa.raster._
+import org.locationtech.geomesa.raster.index.RasterIndexSchema
+import org.locationtech.geomesa.raster.util.RasterUtils
+import org.locationtech.geomesa.security.AuthorizationsProvider
+import org.locationtech.geomesa.utils.geohash.BoundingBox
+import org.locationtech.geomesa.utils.stats.{MethodProfiling, NoOpTimings, Timings, TimingsImpl}
+
+import scala.collection.JavaConversions._
+
+class AccumuloRasterStore(val connector: Connector,
+                  val tableName: String,
+                  val authorizationsProvider: AuthorizationsProvider,
+                  val writeVisibilities: String,
+                  writeMemoryConfig: Option[String] = None,
+                  writeThreadsConfig: Option[Int] = None,
+                  queryThreadsConfig: Option[Int] = None,
+                  collectStats: Boolean = false) extends MethodProfiling with StatWriter with Logging {
+  val writeMemory = writeMemoryConfig.getOrElse("10000").toLong
+  val writeThreads = writeThreadsConfig.getOrElse(10)
+  val bwConfig: BatchWriterConfig =
+    new BatchWriterConfig().setMaxMemory(writeMemory).setMaxWriteThreads(writeThreads)
+  val numQThreads = queryThreadsConfig.getOrElse(20)
+
+  // TODO: WCS: GEOMESA-585 Add ability to use arbitrary schemas
+  val schema = RasterIndexSchema
+  lazy val queryPlanner: AccumuloRasterQueryPlanner = new AccumuloRasterQueryPlanner
+
+  private val tableOps       = connector.tableOperations()
+  private val securityOps    = connector.securityOperations
+  private val profileTable   = s"${tableName}_queries"
+  private def getBoundsRowID = tableName + "_bounds"
+
+  def getAuths = authorizationsProvider.getAuthorizations
+
+  def getMosaicedRaster(query: RasterQuery, params: GeoMesaCoverageQueryParams) = {
```

Comment:
[picky] a one or two line comment here for the method would be great, since this is a big part of the raster store. Something along the lines of "parts of the coverage are spread throughout accumulo, so we reassemble it here"